app/code/community/Satispay/Satispay/controllers/RedirectController.php

```diff
@@ -16,11 +16,16 @@ public function indexAction() {
       \SatispayGBusiness\Api::setPublicKey($helper->getPublicKey());
       \SatispayGBusiness\Api::setPrivateKey($helper->getPrivateKey());
       \SatispayGBusiness\Api::setKeyId($helper->getKeyId($sandbox));
-      $payment = \SatispayGBusiness\Payment::get($this->getRequest()->getQuery('payment_id'));
+      $satispayPayment = \SatispayGBusiness\Payment::get($this->getRequest()->getQuery('payment_id'));
 
-    $order = Mage::getModel('sales/order')->load($payment->metadata->order_id);
+    $order = Mage::getModel('sales/order')->load($satispayPayment->metadata->order_id);
 
-    if ($payment->status === 'ACCEPTED') {
+    if ($satispayPayment->status === 'ACCEPTED') {
+      $payment = $order->getPayment();
+      if (isset($payment)) {
+          $payment->setLastTransId($satispayPayment->id);
+          $payment->save();
```

Comment:
@MichaelBattistonBitbull can you please surround with a try and catch the payment save from row 24 included so if an error is thrown the redirect still occures.